### ISSUE TITLE
Feature Request. Prev and Next Node.

### ISSUE THREAD

**Issue by Quehnie** [sulu/sulu#3308]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | yes

#### Feature Request

Is there a way to get the next node in a twig template. Some extension like sulu_node_next() and sulu_node_prev() would be great.

**Comment by danrot** [comment]
That's a duplicate of https://github.com/sulu/sulu-standard/issues/276, but I will close the other one, since this one is in the correct repository.

**Comment by VnK67** [comment]
Hi, I'm pretty new using Sulu, is this feature out yet ? Thank You

**Comment by danrot** [comment]
No, there is no such feature yet... What we've done already to work around this is to add pages to the navigation, load the navigation, and get the next page from there... But that is a bit cumbersome because the content manager has to do that manually.

**Comment by VnK67** [comment]
Ok thank you. 
I'm trying to doing to map it with a custom controller.
It seems to work for a single content ("firstContent/article"), but I still have difficulties for content nested in other content ("firstContent/Second Content/article).
Work in Progress

**Comment by rskuipers** [comment]
> No, there is no such feature yet... What we've done already to work around this is to add pages to the navigation, load the navigation, and get the next page from there... But that is a bit cumbersome because the content manager has to do that manually.

For those wanting to know how that's done in Twig:

```twig
{% set previousItem = null %}
{% set displayNext = false %}
{% for item in sulu_navigation_root_flat('articles', 2) %}
    {% if item.id == id %}
        {% if previousItem is not null %}
            <a href="{{ sulu_content_path(previousItem.url, previousItem.webspaceKey) }}" title="{{ previousItem.title }}" class="previous">&laquo; {{ previousItem.title }}</a>
        {% endif %}
        {% set displayNext = true %}
    {% elseif displayNext %}
        <a href="{{ sulu_content_path(item.url, item.webspaceKey) }}" title="{{ item.title }}" class="next">{{ item.title }} &raquo;</a>
        {% set displayNext = false %}
    {% endif %}
    {% set previousItem = item %}
{% endfor %}
```

**DISCLAIMER** This code is questionable, but it works for my use-case and since it's my personal blog, I'll allow it :D
